woocommerce/woocommerce
# My subscriptions error notices

### Changes proposed in this Pull Request:

This PR is part of the My subscriptions feature branch.

It introduces standardized success and error notices.
To implement notices we are adding a store and helper functions for adding and removing notices based on the product key.

Error notices are shown on top of the subscriptions list and success notices are shown as snackbars.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the [new subscriptions page](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions)
2. Test install, update, and connect
3. Confirm that all return success snackbars
4. To test errors add the code bellow to the `connectProduct`, `installProduct`, and `updateProduct` methods
```
	return new Promise( ( resolve, reject ) =>
		setTimeout(
			() =>
				reject( {
					success: false,

					data: {
						message: 'Failed',
					},
				} ),

			1000
		)
	);
```
5. Test install, update, and connect
6. Confirm that all return error notices



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Add My subscriptions error notices and move existing notices to helper functions.

</details>
